components/director/internal/domain/scenarioassignment/engine.go

```diff
@@ -66,16 +70,32 @@ func (e *engine) upsertMergedScenarios(ctx context.Context, labels []model.Label
 	return nil
 }
 
-func (engine) RemoveAssignedScenario(in model.AutomaticScenarioAssignment) error {
-	// TODO: Implement it
+func (e *engine) uniqueScenarios(scenarios []interface{}, newScenario string) ([]string, error) {
+	set := make(map[string]struct{})
+	set[newScenario] = struct{}{}
 
-	// remove scenario from runtimes, which have label matching selector
-	return nil
+	for _, scenario := range scenarios {
+		output, ok := scenario.(string)
+		if !ok {
+			return nil, errors.New("scenario is not a string")
+		}
+		set[output] = struct{}{}
+	}
+
+	return str.MapToSlice(set), nil
 }
 
-func (engine) RemoveAssignedScenarios(in []*model.AutomaticScenarioAssignment) error {
-	// TODO: Implement it
+func (e *engine) scenariosDiff(scenarios []interface{}, diffScenario string) ([]string, error) {
```

Comment:
this is basically appending new Scenario, right?
IF so:

```suggestion
func (e *engine) appendScenario(scenarios []interface{}, scenario string) ([]string, error) {
```

Reply:
Nope, It will remove the scenario.
I will rename it to `removeScenario`

Reply:
okay